src/widgets/settingspages/PluginsPage.cpp

```diff
@@ -231,6 +232,13 @@ void PluginsPage::rebuildContent()
         {
             reloadButton->setEnabled(false);
         }
+
+        auto *replButton = new QPushButton("Open REPL", this->dataFrame_);
+        QObject::connect(replButton, &QPushButton::clicked, [id]() {
+            auto *repl = new PluginRepl(id);
+            repl->show();
+        });
+        pluginEntry->addRow(replButton);
```

Comment:
My first instinct is to only have this button available when Chatterino is compiled in debug mode.

I think I'd prefer that to start, and we can think of how we can safely allow this in non-debug modes later, maybe something like
Extra run flag? `chatterino.exe --enable-lua-repl`
Environment variable? `CHATTERINO_ENABLE_LUA_REPL=1`
Hidden setting in `settings.json` that you can only enable by manually editing settings.json? `{"I_KNOW_WHAT_IM_DOING": true}`

Reply:
I would like to have them be available in release builds without too much effort. The goal was to allow developers, also people who just started, to easily debug their plugins. I could see the following: unconditionally enable in debug mode, enable if the author name matches the current user, always enable if a hidden setting is enabled.

Reply:
I'd like the REPL to be completely disabled unless a user has started chatterino with the `--enable-lua-repl` flag (I'm ok with different naming)

Reply:
imo if we must hide it, we should do the hidden setting. It's easy enough to explain to other programmers and persistent so a user won't need to enable it every time the use the program which is a pain in the ass

Reply:
What I don't like with it is how easy it is to just brick your settings, but I am ok with it